src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserHelperClasses.cs

```diff
@@ -300,103 +300,40 @@ internal void Clear()
 
     internal static class SqlEnvChangePool
```

Comment:
A comment about the behavior of this class would be nice.

Reply:
Added, and a couple of others inline to explain why items are new'ed and possibly dropped to the GC. Let me know if that explains things better and what else might be useful.

Reply:
Comma should be split into two sentences. (http://www.sussex.ac.uk/informatics/punctuation/comma/joining)
Sentences should consistently have a capital letter and an end mark.